gateway-service/src/main/java/com/example/gatewayservice/filter/IsClientCheckFilter.java

```diff
@@ -1,4 +1,54 @@
 package com.example.gatewayservice.filter;
 
-public class IsClientCheckFilter {
+
+import com.example.gatewayservice.common.exception.BusinessException;
+import com.example.gatewayservice.common.exception.ErrorCode;
+import com.example.gatewayservice.common.jwt.JwtTokenParser;
+import io.jsonwebtoken.Claims;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.cloud.gateway.filter.GatewayFilter;
+import org.springframework.cloud.gateway.filter.factory.AbstractGatewayFilterFactory;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class IsClientCheckFilter extends
+    AbstractGatewayFilterFactory<IsClientCheckFilter.Config> {
+
+    private static final Set<String> ALLOWED_ROLES = Set.of("CLIENT", "BOTH");
```

Comment:
아래와 같이 작성하면 하드 코딩을 피할 수 있을 것 같습니다.

```suggestion
    private static final Set<String> ALLOWED_ROLES = Set.of(
        MemberRole.CLIENT.name(),
        MemberRole.BOTH.name()
    );
```

<br>

추가로, 역할 간 포함 관계가 있다 보니 (CLIENT ⊂ BOTH), 각 역할에 비트 값을 부여하고 AND 연산으로 검증하는 방법도 편리하지 않을까 싶어요!

```java
enum Role {
    NONE(0b000),
    CLIENT(0b001),
    BOTH(0b011),  // CLIENT 권한 포함

    private final int mask;

    Role(int mask) {
        this.mask = mask;
    }

    public int getMask() {
        return mask;
    }

    public boolean includes(Role other) {
        return (this.mask & other.mask) == other.mask;
    }
}
```

Reply:
하드 코딩과 관련해서 고민을 하였었는데
확장성적인 면에서는 하드 코딩보다 Enum으로 관리하는 것이 더 좋을 것이라고 생각합니다!

이 때
Member 모듈의 MemberRole을 Core로 빼기에는 
MemberRole을 몰라도 되는 다른 (member, gateway를 제외한) 모듈까지 core를 의존할 때 포함되게 하는 것이 애매하다고 생각하였고, 

Gateway 모듈에서 따로 enum으로 개발 시, 
변경이 일어나지 않겠지만 변경이 일어날 경우 Enum과 Filter 모두에서 변경이 일어나야할 것 같아

일단은 확장을 고려하지 않은 상태에서 관리를 하기에는 하드 코딩으로 하는 것이 더 편할 것 같아 PR과 같이 작성을 하였습니다.